### ISSUE TITLE
NN Assignment should apply to all installs on a connected building

### ISSUE THREAD

**Issue by Andrew-Dickinson** [nycmeshnet/meshdb#787]
When applying an NN to an install, we should also apply the NN to all installs in the associated building object

**Comment by Andrew-Dickinson** [comment]
We also need to backfill or at least detect any times where this didn't happen and should have and manually correct them

**Comment by Andrew-Dickinson** [comment]
https://nycmesh.slack.com/archives/C06GW3Q92TY/p1734784142508089

**Comment by WillNilges** [comment]
This is for events where we have multiple installs in a building, do all of them, but only submit the NN assign form once? What if we have multiple installs in a building but _don't_ install all of them?

Maybe we need a checkbox that's like, "Apply NN to all installs in this Building".

As for backfills, how will we know that the installs are actually connected?

**Comment by Andrew-Dickinson** [comment]
Ope wow I fixed this today and forgot to check for a GH issue haha

I think perhaps you're misunderstanding the Install <-> Node relation. A pairing between them doesn't mean that the install is completed, for potential installs, it just means that is the node that the install _would_ use if it were to get connected

**Comment by Andrew-Dickinson** [comment]
I disagree that it should be opt in (or even opt out). IMO it is incorrect to have non-node-ed potential installs dangling off the building object when the building has a primary node

**Comment by Andrew-Dickinson** [comment]
https://nycmesh.slack.com/archives/C06GW3Q92TY/p1737487981435119

**Comment by WillNilges** [comment]
OK so just because a node is linked to an install doesn't mean that install is necessarily going to be marked as active. That makes sense. That way when you go to do the install, or are planning some event like an install-a-thon, you can clearly see "oh this install is literally just a cable run."

**Comment by Andrew-Dickinson** [comment]
Yep, and in fact OSTicket is programmed to do that, and sends a different message based on this info
